core/esmf-aspect-model-aas-generator/src/main/java/org/eclipse/esmf/aspectmodel/aas/AspectModelAASVisitor.java

```diff
@@ -558,102 +450,156 @@ public Environment visitCharacteristic(
+   private <T extends Collection> List<SubmodelElement> getValues( final T collection, final Property property, final Context context,
+         final ArrayNode arrayNode ) {
+      return collection.getDataType()
+            .map( dataType -> {
+               if ( Scalar.class.isAssignableFrom( dataType.getClass() ) ) {
+                  return List.of( (SubmodelElement) new DefaultBlob.Builder().value( StreamSupport.stream( arrayNode.spliterator(), false )
+                        .map( JsonNode::asText )
+                        .collect( Collectors.joining( "," ) )
+                        .getBytes( StandardCharsets.UTF_8 ) ).build() );
+               } else {
+                  final var values = StreamSupport.stream( arrayNode.spliterator(), false )
+                        .map( n -> {
+                           context.iterate( property );
+                           return decideOnMapping( property, context );
+                        } )
+                        .toList();
+                  context.finishIteration( property );
+                  return values;
+               }
+            } ).orElseGet( () -> List.of() );
    }
 
    // Either will be mapped by adding both the left and the right side to the SubmodelTemplate.
-   // No possibilities to mark the SubmodelElements as optional. So both are either options are
+   // No possibilities to mark the SubmodelElements as optional. So both either options are
    // in the result and have to be manually selected.
+   // When generating Submodels where data is given however only the present side is added.
    @Override
    public Environment visitEither(
          final Either either, final Context context ) {
       final List<SubmodelElement> submodelElements = new ArrayList<>();
-      if ( either.getLeft().getDataType().isPresent() ) {
-         submodelElements.add(
-               decideOnMapping( either.getLeft().getDataType().get(), context.getProperty(), context ) );
-      }
-      if ( either.getRight().getDataType().isPresent() ) {
-         submodelElements.add(
-               decideOnMapping( either.getRight().getDataType().get(), context.getProperty(), context ) );
-      }
-      final SubmodelElementList aasSubModelElementCollection =
+      final var property = context.getProperty();
+      either.getLeft().getDataType()
+            .flatMap( dataType -> handleEitherField( "left", either.getLeft(), property, context ) )
+            .ifPresent( submodelElements::add );
+      either.getRight().getDataType()
+            .flatMap( dataType -> handleEitherField( "right", either.getRight(), property, context ) )
+            .ifPresent( submodelElements::add );
+
+      final SubmodelElementList eitherSubModelElements =
             new DefaultSubmodelElementList.Builder()
                   .idShort( ID_PREFIX + either.getName() )
                   .typeValueListElement( AASSubmodelElements.DATA_ELEMENT ) // TODO check if more specific type info is rqujried
```

Comment:
Can this TODO be addressed? Also, there is a typo (a different one than in the TODO comment above :-))

Reply:
Same, DATA_ELEMENT should be fine.